spikeinterface/curation/tests/test_remove_excess_spikes.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+from spikeinterface.core.generate import generate_recording, NumpySorting
+from spikeinterface.core.waveform_tools import has_exceeding_spikes
+from spikeinterface.curation import remove_excess_spikes
+
+
+def test_remove_excess_spikes():
+    durations = [5, 4]
+    sampling_frequency = 30000
+    recording = generate_recording(durations=durations, sampling_frequency=sampling_frequency)
+
+    # create two units with excess 5 excess spikes each in each segment
+    num_units = 2
+    num_spikes = 100
+    num_excess_spikes_per_segment = 5
+    times = []
+    labels = []
+    for segment_index in range(recording.get_num_segments()):
+        num_samples = recording.get_num_samples(segment_index=segment_index)
+        times_segment = np.array([], dtype=int)
+        labels_segment = np.array([], dtype=int)
+        for unit in range(num_units):
+            spike_times = np.random.randint(0, num_samples - 1, num_spikes)
+            excess_spikes = np.random.randint(num_samples, num_samples + 100, num_excess_spikes_per_segment)
```

Comment:
Instead of random I would force hard coded corner case like exactly "num_samples" with a manul list.